src/stores/models/AddressBalance.test.js

```diff
@@ -0,0 +1,54 @@
+import AddressBalance from './AddressBalance';
+import { decimalToSatoshi } from '../../helpers/utility';
+
+const localStorageMock = {
+  getItem: jest.fn(),
+  setItem: jest.fn(),
+  clear: jest.fn(),
+};
+const navigatorMock = {};
+global.navigator = navigatorMock;
+global.localStorage = localStorageMock;
+
+describe('stores/models/AddressBalance', () => {
+  let ab;
+  beforeEach(() => ab = null);
+  it('Constructor input Satoshi', () => {
+    const input = {
+      address: '123456',
+      qtum: decimalToSatoshi(new Number(10)), // eslint-disable-line
+      bot: decimalToSatoshi('2000'),
+    };
+    ab = new AddressBalance(input);
+
+    expect(ab.address).toBe(input.address);
+    expect(ab.qtum).toBe(10);
+    expect(ab.bot).toBe(2000);
+  });
+
+  it('Constructor input floating', () => {
+    const input = {
+      address: '123456',
+      qtum: decimalToSatoshi(new Number(10.5)), // eslint-disable-line
+      bot: decimalToSatoshi('2000.5'),
+    };
+    ab = new AddressBalance(input);
+
+    expect(ab.address).toBe(input.address);
+    expect(ab.qtum).toBe(10.5);
+    expect(ab.bot).toBe(2000.5);
+  });
+
+  it('Constructor input NAN', () => {
```

Comment:
ok to leave this now, but we should add validation so non-numbers should not be accepted. this is a good example where writing a test leads to changing the code in a better way.